d2l-button-toggle.js

```diff
@@ -0,0 +1,154 @@
+import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
+import 'd2l-offscreen/d2l-offscreen.js';
+import './d2l-button-shared-styles.js';
+
+/**
+ * @customElement
+ * @polymer
+ */
+class D2LButtonToggle extends PolymerElement {
+	static get is() { return 'd2l-button-toggle'; }
+	static get template() {
+		return html`
+		<style>
+			:host([disabled]) .d2l-button-toggle-button {
+				opacity: 0.5;
+				cursor: default;
+			}
+			.d2l-button-toggle-container > li:first-of-type > button {
+				border-bottom-left-radius: 0.3rem;
+				border-top-left-radius: 0.3rem;
+				border-left-width: 1px;
+			}
+			:host(:dir(rtl)) .d2l-button-toggle-container > li:first-of-type > button {
+				border-bottom-right-radius: 0.3rem;
+				border-top-right-radius: 0.3rem;
+			}
+			.d2l-button-toggle-container > li:last-of-type > button {
+				border-bottom-right-radius: 0.3rem;
+				border-top-right-radius: 0.3rem;
+			}
+			:host(:dir(rtl)) .d2l-button-toggle-container > li:last-of-type > button {
+				border-bottom-left-radius: 0.3rem;
+				border-top-left-radius: 0.3rem;
+				border-left-width: 1px;
+			}
+			.d2l-button-toggle-button,
+			.d2l-button-toggle-button[aria-pressed] {
+				@apply --d2l-button;
+				@apply --d2l-button-shared;
+				@apply --d2l-label-text;
+				border-radius: 0;
+				border-width: 1px 1px 1px 0px;
+				font-family: inherit;
+				padding: 0.5rem 1.5rem;
+			}
+			.d2l-button-toggle-button {
+				background-color: var(--d2l-color-regolith);
+			}
+			:host(:not([disabled])) .d2l-button-toggle-button:hover,
+			:host(:not([disabled])) .d2l-button-toggle-button:focus,
+			.d2l-button-toggle-button[aria-pressed]
+			{
+				background-color: var(--d2l-color-gypsum);
+			}
+			.d2l-button-toggle-button:focus,
+			.d2l-button-toggle-button[aria-pressed]:focus {
+				box-shadow: 0 0 1px 0 var(--d2l-color-sylvite) inset, 0 0 3px 1px #99C5E5;
+				position: relative;
+			}
+			.d2l-button-toggle-container {
+				list-style: none;
+				white-space: no-wrap;
+				display: inline-block;
+			}
+			.d2l-button-toggle-container li {
+				display: inline-block;
+			}
+		</style>
+		<d2l-offscreen><h2 id="d2l-button-toggle-title">[[title]]</h2></d2l-offscreen>
+		<ul class="d2l-button-toggle-container" role="toolbar" aria-labelledby="d2l-button-toggle-title">
+			<dom-repeat items="[[buttons]]" as="b">
```

Comment:
I could be convinced otherwise, but I prefer an API that is simply HTML and that separates the concerns of the container/list from the actual button:
```
<d2l-button-toggles label="...">
   <d2l-button-toggle key="..." text="..."></d2l-button-toggle>
   <d2l-button-toggle key="..." text="..." selected></d2l-button-toggle>
   <d2l-button-toggle key="..." text="..." disabled></d2l-button-toggle>
   ...
</d2l-button-toggles>
```
In such case, it should also work if a button is hidden via `hidden` attribute.